java/arcs/core/data/SingletonType.kt

```diff
@@ -24,12 +24,16 @@ import kotlin.reflect.KClass
 data class SingletonType<T : Type>(override val containedType: T) :
     Type,
     Type.TypeContainer<T>,
+    EntitySchemaProviderType,
     CrdtModelType<
         CrdtSingleton.Data<Referencable>,
         CrdtSingleton.IOperation<Referencable>,
         Referencable?> {
     override val tag = Tag.Singleton
 
+    override val entitySchema: Schema?
+        get() = (containedType as? EntitySchemaProviderType)?.entitySchema
```

Comment:
since containedType is final, this doesn't need to be a getter, it can just be a normal field
```suggestion
    override val entitySchema: Schema? = 
        (containedType as? EntitySchemaProviderType)?.entitySchema
```